stactools_pipelines/pipelines/noaa_hrrr/app.py

```diff
@@ -0,0 +1,44 @@
+import json
+import os
+
+import requests
+from aws_lambda_powertools.utilities.data_classes import SQSEvent, event_source
+from stactools.core import use_fsspec
+from stactools.noaa_hrrr.constants import COLLECTION_ID_FORMAT
+from stactools.noaa_hrrr.metadata import parse_href
+from stactools.noaa_hrrr.stac import create_item
+
+from stactools_pipelines.cognito.utils import get_token
+
+
+@event_source(data_class=SQSEvent)
+def handler(event: SQSEvent, context):
+    ingestor_url = os.environ["INGESTOR_URL"]
+    ingestions_endpoint = f"{ingestor_url.strip('/')}/ingestions"
+    token = get_token()
+    headers = {"Authorization": f"bearer {token}"}
+    use_fsspec()
+    for record in event.records:
+        record_body = json.loads(record.body)
+        record_message = json.loads(record_body["Message"])
+        for sns_record in record_message["Records"]:
+            key = sns_record["s3"]["object"]["key"]
+            path = f"https://noaa-hrrr-bdp-pds.s3.amazonaws.com/{key.lstrip('/')}"
+            print(path)
+            href_parsed = parse_href(path)
+            stac = create_item(**href_parsed)
+
+            stac.collection_id = COLLECTION_ID_FORMAT.format(
```

Comment:
This whole multi-cloud provider support you built into the `stactools` package is really nice.  We should see if we can upstream some of this into recommendations for the core `stactools` template.

Reply:
Yeah! It is definitely relevant the numerous datasets that are hosted on all of the cloud platforms since the key structure is typically identical except for the domain/bucket prefix. Maybe it could go somewhere in the [stactools](https://github.com/stac-utils/stactools) package so that it could be easily re-used in sub-packages.